### PR TITLE
Feature/22 download node npm

### PR DESCRIPTION
**Pull request by mvysny**
See #22 for more details.

### CODE REVIEW

**Comment by mvysny** [comment]
Fixed: now also the base-starter-spring-gradle project downloads the node distro correctly. @mstahv could you please review & merge?

File: README.md [hunk 1]
```diff
@@ -100,6 +102,37 @@ for more details.
 * `optimizeBundle = true`: Whether to use byte code scanner strategy to discover frontend
   components.
 
+## Automatic Download of node.js and npm
+
+You do not need to have node.js nor npm installed in your system, in order to use Vaadin.
+The `vaadinPrepareNode` task will download the node.js+npm distribution and will place it
+into the `node/` folder which will then be picked automatically, both by Vaadin and
```
Comment (mstahv) [inline README.md]:
```suggestion
into the `node/` folder which will then be picked automatically by Vaadin.
```

File: README.md [hunk 2]
```diff
@@ -100,6 +102,37 @@ for more details.
...
+the Vaadin Gradle Plugin.
```
Comment (mstahv) [inline README.md]:
```suggestion
```

File: README.md [hunk 3]
```diff
@@ -100,6 +102,37 @@ for more details.
...
+In your CI, don't forget to call the `vaadinPrepareNode` before the `vaadinPrepareFrontend` task:
+```bash
+./gradlew clean vaadinPrepareNode vaadinBuildFrontend build
+```
+
+If you wish to override the node version which will be downloaded, simply specify
+the node.js version in the `vaadinFlow{}` block:
+
+```groovy
+vaadinFlow {
```
Comment (mstahv) [inline README.md]: Let's not use `vaadinFlow`, but just `vaadin`.

File: src/main/kotlin/com/vaadin/gradle/VaadinPrepareNodeTask.kt [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+import org.gradle.api.DefaultTask
+import org.gradle.api.tasks.TaskAction
+import java.io.File
+import java.lang.IllegalStateException
+
+/**
+ * Copies the node.js distro downloaded by the com.github.node-gradle.node plugin
+ * to the node/ folder which Vaadin expects.
+ *
+ * You need to manually include the com.github.node-gradle.node plugin into your build file
```
Comment (mstahv) [inline src/main/kotlin/com/vaadin/gradle/VaadinPrepareNodeTask.kt]: It is bit unclear for me now, is this needed or not? If needed, then we should mention that in README.md as well.

**Pull request merged**